cirq/experiments/random_quantum_circuit_generation.py

```diff
@@ -217,6 +236,114 @@ def random_rotations_between_two_qubit_circuit(
     return circuit
 
 
+def generate_library_of_2q_circuits(
+    n_library_circuits: int,
+    two_qubit_gate: 'cirq.Gate',
+    *,
+    max_cycle_depth: int = 100,
+    q0: 'cirq.Qid' = devices.LineQubit(0),
+    q1: 'cirq.Qid' = devices.LineQubit(1),
+    random_state: 'cirq.RANDOM_STATE_OR_SEED_LIKE' = None,
+) -> List['cirq.Circuit']:
+    """Generate a library of two-qubit Circuits.
+
+    Args:
+        n_library_circuits: The number of circuits to generate.
+        two_qubit_gate: The two qubit gate to use in the circuits.
+        max_cycle_depth: The maximum cycle_depth in the circuits to generate. If you are using XEB,
+            this must be greater than or equal to the maximum value in `cycle_depths`.
+        q0: The first qubit to use when constructing the circuits.
+        q1: The second qubit to use when constructing the circuits
+        random_state: A random state or seed used to deterministically sample the random circuits.
+    """
+    rs = value.parse_random_state(random_state)
+    exponents = np.linspace(0, 7 / 4, 8)
+    single_qubit_gates = [
+        ops.PhasedXZGate(x_exponent=0.5, z_exponent=z, axis_phase_exponent=a)
+        for a, z in itertools.product(exponents, repeat=2)
+    ]
+    return [
+        random_rotations_between_two_qubit_circuit(
+            q0,
+            q1,
+            depth=max_cycle_depth,
+            two_qubit_op_factory=lambda a, b, _: two_qubit_gate(a, b),
+            single_qubit_gates=single_qubit_gates,
+            seed=rs,
+        )
+        for _ in range(n_library_circuits)
+    ]
+
+
+def _get_active_pairs(graph: nx.Graph, grid_layer: GridInteractionLayer):
+    """Extract pairs of qubits from a device graph and a GridInteractionLayer."""
+    for edge in graph.edges:
+        if edge in grid_layer:
+            yield edge
+
+
+@dataclasses.dataclass(frozen=True)
+class CircuitLibraryCombination:
+    """For a given layer (which defines a set of pairs of qubits), `combinations` is a 2d array
+    of shape (n_combinations, len(pairs)) where each row represents a combination (with replacement)
```

Comment:
n_combinations is number of "experiments" which are used for fidelity estimation? The more of them the better accuracy?

Reply:
technically more "precise". But yes, it's the number of random circuits. I've added a note to `get_random_combinations_for_device` that increasing `n_combinations` increases precision of the estimate but increases sampling runtime

Reply:
Sounds good, I just started to wonder if n_combinations is a good name. Combinations is not the main concern in the end (they could have been generated on the fly as well), it's number of circuit sampled what really matters.